.github/workflows/ui.yml

```diff
@@ -1,17 +1,18 @@
 name: ui - build and deploy
 
 on:
+  pull_request:
+    paths:
+      - "ui/**"
+      - "package-lock.json"
+      - ".github/workflows/ui.yml"
   push:
+    branches:
```

Comment:
@andrashee 

> only pushes to main or PRs are triggers

I'm pretty sure this is not what it does. Based on the workflow documentation of GitHub, the `push` rule here will now run **when at minimum one of both definitions in the push rule applies** (not _when at least both apply_):

- When someone pushes to main
- When there are changes in the defined paths

Based on that, this rule change would make that any changes on the main branch (also in the admin folder or on the readme) always trigger the workflow.

Prove me wrong, but that was actually what I always found when I looked up how to make this an **AND** rule.

Reply:
The
```
branches:
   - main
paths:
   - xyz
```
is definitely an AND condition. We are using the same pattern in the other workflows.

The documentation is not so clear what is chained as an OR or AND. Here someone had the same question: https://stackoverflow.com/questions/63460875/github-actions-how-do-you-trigger-a-push-when-a-specific-directory-in-a-branch

Top-level (pull_request, push) are ORs.

Reply:
@renestalder, wdyt shall we try this?

Reply:
:+1: